### DISCUSSION TITLE
Parallel Routes not working on hard navigation

### DISCUSSION THREAD

**Discussion by danikova** [vercel/next.js#52951] [App Router]
Hi! So I have a side project and I have this parallel route folder setup in it:
<img width="148" alt="image" src="https://github.com/vercel/next.js/assets/30092494/2706b9a5-25f8-4d55-855f-b7bb10cfd777">

- I put everywhere a default file to work with hard navigation but for some reason it's not working, in other hand for soft navigation with the _Link_ componet is working fine
- maybe I missed something during the setup

here is my layout file:
<img width="611" alt="image" src="https://github.com/vercel/next.js/assets/30092494/035a1788-2aa2-4e58-9496-f30ff653c61d">

here is the base view:
<img width="793" alt="image" src="https://github.com/vercel/next.js/assets/30092494/ad398b1b-3455-4195-ace8-1eef236580f6">

soft navigation working with _Link_:
<img width="798" alt="image" src="https://github.com/vercel/next.js/assets/30092494/27206811-e519-42ec-9692-2da6a0e011ef">

hard navigation not working after refresh:
<img width="853" alt="image" src="https://github.com/vercel/next.js/assets/30092494/d658ec0a-98b3-46b8-a532-3cef62910c58">

**Comment by akomm** [comment]
Despite parallel routes being utterly broken currently (alpha stage unofficially), I see a problem in your case that is not based on a bug.

layout.tsx does nest, but page.tsx does not. Only the top most page.tsx is loaded. 

Example - if you are in `/results/details`, you load the following layouts, if the file exist:
/layout.tsx
/results/layout.tsx
/results/details/layout.tsx

BUT, you will only load the page:
/results/details/page.tsx

If you do not have page.tsx there, its 404

Same applies for hard loaded parallel routes.

Every slot in your layout, `detail` and `children` has to match a page.tsx

That means if you are in `/results/details` you match the url of the page.tsx within, BUT you do not match the page of `/results` which is your `children` slot - remember, url matches any layout.tsx on the way down, but only the top most page.tsx, so your one level higher /results/page.tsx does not match. However /results/layout.tsx wants to render the children slot. Since the page does not match, its 404.

To fix the issue, you need to provide a default for that route segment:
/results/default.tsx

you can re-export your page from within default.tsx:
```ts
import ResultsPage from "@/results/page"

export default ResultsPage
```

Just think of any slot requires the page URL to match, if you want to allow it to not match, provide a default export with whatever you want to be rendered in the slot.

**Comment by ProSystemLimited** [comment]
```
app
└── patients
    ├── @modal
    │   ├── [patientId]
    │   │   ├── @prescription
    │   │   │   ├── prescription
    │   │   │   │   └── page.js
    │   │   │   ├── layout.js
    │   │   │   └── default.js
    │   │   ├── layout.js
    │   │   └── page.js
    │   ├── new
    │   │   └── page.js
    │   └── default.js
    ├── layout.js
    └── page.js
```
I am facing a problem where hard navigation to parallel routes show 404 page not found but navigating via link component is working as expected.

The parallel routes are displayed on a full screen dialog which overlays over the main patients page. When navigating to /patients/new, /patients/123, /patients/123/prescription via link component, it is working as expected but when hard navigating to any of the links, I am getting 404 page not found, however, I want the parallel route to be displayed upon hard navigation just as it does on soft navigation.

Can this be done using parallel routes? Really need help on this one

**Reply by akomm** [reply]
I've dropped using next back then, so things might have changed, but:

Which slots are used in `parients/layout.tsx`?
Each slot used needs to match a page.tsx all times or have a default.tsx with an export as fallback provided for *hard reload*.

For detailed explanation just read my above [reply](https://github.com/vercel/next.js/discussions/52951#discussioncomment-6852364).
